lib/plugins/nwc/src/persist.rs

```diff
@@ -26,30 +31,113 @@ impl Persister {
         self.storage.get_item(KEY_NWC_SECKEY).map_err(Into::into)
     }
 
-    pub(crate) fn set_nwc_uri(&self, name: String, uri: String) -> NwcResult<()> {
-        let mut nwc_uris = self.list_nwc_uris()?;
-        nwc_uris.insert(name, uri);
+    pub(crate) fn add_nwc_connection(
+        &self,
+        name: String,
+        connection: NwcConnection,
+    ) -> NwcResult<()> {
+        let mut connections = self.list_nwc_connections()?;
+        if connections.contains_key(&name) {
+            return Err(NwcError::generic(format!(
+                "Could not insert connection: `{name}` already exists"
+            )));
+        }
+        connections.insert(name, connection);
+        self.storage
+            .set_item(KEY_NWC_CONNECTIONS, serde_json::to_string(&connections)?)?;
+        Ok(())
+    }
```

Comment:
Could we get races here that cause one of the connections to end up not persisted?

Reply:
Same with `edit_nwc_connection` and `remove_nwc_connection`. Maybe this has been discussed in the past but I'm missing why it's safe.